### PR TITLE
Refactor: Moving disk-related cloud provider operations to gce_disks.go

### LINKED ISSUE #51280: Move disk-related operations out of gce.go
The main GCE cloud provider code (pkg/cloudprovider/providers/gce/gce.go) should not contain disk-related operations. Move them to gce_disks.go

/cc @bowei 
/assign

### PR DESCRIPTION
**Pull request by verult**
**What this PR does / why we need it**: The main GCE cloud provider code (pkg/cloudprovider/providers/gce/gce.go) should not contain disk-related operations. Moved them to gce_disks.go

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #51280 

**Release note**:

```release-note
NONE
```
/release-note-none
/sig storage
/assign @msau42 @bowei

### CODE REVIEW

File: pkg/cloudprovider/providers/gce/gce_disks.go [hunk 1]
```diff
@@ -43,6 +43,52 @@ const (
 	diskTypeUriTemplate = "%s/zones/%s/diskTypes/%s"
 )
 
+type DiskServiceManager interface {
```
Comment (bowei) [inline pkg/cloudprovider/providers/gce/gce_disks.go]: this does not need to be exported `diskServiceManager`

File: pkg/cloudprovider/providers/gce/gce_disks.go [hunk 2]
```diff
@@ -43,6 +43,52 @@ const (
...
+	GetDiskOnCloudProvider(project string, zone string, diskName string) (*compute.Disk, error)
+
+	// Deletes the persistent disk from GCE with the given diskName.
+	DeleteDiskOnCloudProvider(project string, zone string, disk string) (*compute.Operation, error)
+
+	// Waits until GCE reports the given operation in the given zone as done.
+	WaitForZoneOp(op *compute.Operation, zone string, mc *metricContext) error
+}
+
+type GCEServiceManager struct {
```
Comment (bowei) [inline pkg/cloudprovider/providers/gce/gce_disks.go]: unexport

File: pkg/cloudprovider/providers/gce/gce_disks.go [hunk 3]
```diff
@@ -43,6 +43,52 @@ const (
...
+func (manager *GCEServiceManager) DeleteDiskOnCloudProvider(
+    project string,
+    zone string,
+    diskName string) (*compute.Operation, error) {
+
+	return manager.gce.service.Disks.Delete(project, zone, diskName).Do()
+}
+
+func (manager *GCEServiceManager) WaitForZoneOp(op *compute.Operation, zone string, mc *metricContext) error {
+	return manager.gce.waitForZoneOp(op, zone, mc)
```
Comment (bowei) [inline pkg/cloudprovider/providers/gce/gce_disks.go]: why replicate this?

File: pkg/cloudprovider/providers/gce/gce_disks.go [hunk 4]
```diff
@@ -43,6 +43,52 @@ const (
...
+	WaitForZoneOp(op *compute.Operation, zone string, mc *metricContext) error
+}
+
+type GCEServiceManager struct {
+	gce *GCECloud
+}
+
+func (manager *GCEServiceManager) CreateDiskOnCloudProvider(
+    project string,
+    zone string,
```
Comment (bowei) [inline pkg/cloudprovider/providers/gce/gce_disks.go]: project, zone string

File: pkg/cloudprovider/providers/gce/gce_disks.go [hunk 5]
```diff
@@ -43,6 +43,52 @@ const (
...
+    disk *compute.Disk) (*compute.Operation, error) {
+
+	return manager.gce.service.Disks.Insert(project, zone, disk).Do()
+}
+
+func (manager *GCEServiceManager) GetDiskOnCloudProvider(
+    project string,
```
Comment (bowei) [inline pkg/cloudprovider/providers/gce/gce_disks.go]: project, zone, diskName string

File: pkg/cloudprovider/providers/gce/gce_disks.go (hunk 3 above)
Comment (verult) [inline pkg/cloudprovider/providers/gce/gce_disks.go]: It's called inside the disk methods in gce_disks.go. The unit test doesn't issue any actual GCE operations, so this needs to be mocked too.

**Comment by verult** [comment]
Addressed comments and test failures.

**Comment by verult** [comment]
/retest
pull-kubernetes-e2e-gce-gpu issue: https://github.com/kubernetes/kubernetes/issues/51481

**Comment by saad-ali** [comment]
This PR is approved and LGTM'd for 1.8, but putting the `do-not-merge` label on it to allow PR https://github.com/kubernetes/kubernetes/pull/51667 to merge first. Once that PR merges, we can rebase this PR and let it merge.

**Comment by saad-ali** [comment]
PR #51667 merged. Ok to rebase and merge this. Removing `do-not-merge`

**Comment by verult** [comment]
@saad-ali PTAL, thanks!

**Comment by dims** [comment]
is this too late for 1.8?

**Comment by saad-ali** [comment]
Too late for 1.8. Let's get this in to 1.9 as soon as the code freeze is lifted.

**Pull request merged**